### ISSUE TITLE
[Triage] Attempts to download from any source time out the web UI

### ISSUE THREAD

**Issue by gooeymiddle** [kieraneglin/pinchflat#486]
Recently I have been unable to download any videos from any source I've tried, whether it be a playlist or channel. I had an existing source monitoring a channel that was working great with no issues, up until recently as well. Every time I try a new source, with any media profile, the web UI just times out and nothing happens or beings to download. 

I have the latest version of pinchflat, v2024.10.30, and the latest version of Docker, v4.34.3. I'm running Ubuntu 22.04.4 LTS x86_64. I have tried uninstalling and reinstalling both pinchflat and the docker engine itself, to no avail.

This is the log output from when I try to begin the download of, in this case, a channel

```
2024-11-26 23:06:03 2024-11-27 04:06:03.270 request_id=GAu3nDoIsYqZKm8AAAAB [info] | POST /sources
2024-11-26 23:06:03 2024-11-27 04:06:03.270 request_id=GAu3nDoIsYqZKm8AAAAB [debug] | Processing with PinchflatWeb.Sources.SourceController.create/2
2024-11-26 23:06:03   Parameters: %{"_csrf_token" => "VhQVAF4xFRUsHihYQlsIOAdcIitrLBEh9_vi3U_Tb_R5tijuUmCn8CxS", "source" => %{"custom_name" => "", "download_cutoff_date" => "", "download_media" => "true", "fast_index" => "false", "index_frequency_minutes" => "1440", "max_duration_seconds" => "", "media_profile_id" => "1", "min_duration_seconds" => "", "original_url" => "https://www.youtube.com/@CalebHammer", "output_path_template_override" => "", "retention_period_days" => "", "title_filter_regex" => "", "use_cookies" => "false"}}
2024-11-26 23:06:03   Pipelines: [:browser]
2024-11-26 23:06:03 2024-11-27 04:06:03.272 request_id=GAu3nDoIsYqZKm8AAAAB [info] | [command_wrapper]: /usr/local/bin/yt-dlp called with: https://www.youtube.com/@CalebHammer --simulate --skip-download --ignore-no-formats-error --playlist-end 1 --print-to-file %(.{channel,channel_id,playlist_id,playlist_title,filename})j /tmp/pinchflat/data/e342350b12d56ede70735023e694915da2549388bf591cda629b65d851a86617.json --windows-filenames --quiet --cache-dir /tmp/pinchflat/data/yt-dlp-cache
2024-11-26 23:06:37 2024-11-27 04:06:37.956 request_id=GAu3pE2Cz-8VZdIAAAND [info] | GET /healthcheck
2024-11-26 23:06:37 2024-11-27 04:06:37.956 request_id=GAu3pE2Cz-8VZdIAAAND [debug] | Processing with PinchflatWeb.HealthController.check/2
2024-11-26 23:06:37   Parameters: %{}
2024-11-26 23:06:37   Pipelines: [:api]
2024-11-26 23:06:37 2024-11-27 04:06:37.957 request_id=GAu3pE2Cz-8VZdIAAAND [info] | Sent 200 in 247µs
```

**Comment by adamsih300u** [comment]
I have the same issue, except I'm using Kubernetes. Adding a source with Fast Index or Download Media turned on or off results in a timeout of the UI. This is with a completely fresh install.

My logs leading up to the issue:

```
2024-11-27 18:49:52.697 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [debug] | QUERY OK source="media_profiles" db=0.4ms idle=1534.2ms
SELECT m0."id", m0."name", m0."redownload_delay_days", m0."output_path_template", m0."download_subs", m0."download_auto_subs", m0."embed_subs", m0."sub_langs", m0."download_thumbnail", m0."embed_thumbnail", m0."download_source_images", m0."download_metadata", m0."embed_metadata", m0."download_nfo", m0."sponsorblock_behaviour", m0."sponsorblock_categories", m0."shorts_behaviour", m0."livestream_behaviour", m0."preferred_resolution", m0."marked_for_deletion_at", m0."inserted_at", m0."updated_at" FROM "media_profiles" AS m0 ORDER BY m0."name" []
2024-11-27 18:49:52.697 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [debug] | QUERY OK source="settings" db=0.1ms idle=1535.0ms
SELECT s0."id", s0."onboarding", s0."pro_enabled", s0."yt_dlp_version", s0."apprise_version", s0."apprise_server", s0."youtube_api_key", s0."video_codec_preference", s0."audio_codec_preference" FROM "settings" AS s0 LIMIT 1 []
2024-11-27 18:49:52.698 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [debug] | QUERY OK source="settings" db=0.1ms idle=1535.6ms
SELECT s0."id", s0."onboarding", s0."pro_enabled", s0."yt_dlp_version", s0."apprise_version", s0."apprise_server", s0."youtube_api_key", s0."video_codec_preference", s0."audio_codec_preference" FROM "settings" AS s0 LIMIT 1 []
2024-11-27 18:49:52.699 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [debug] | QUERY OK source="settings" db=0.1ms idle=1317.8ms
SELECT s0."id", s0."onboarding", s0."pro_enabled", s0."yt_dlp_version", s0."apprise_version", s0."apprise_server", s0."youtube_api_key", s0."video_codec_preference", s0."audio_codec_preference" FROM "settings" AS s0 LIMIT 1 []
2024-11-27 18:49:52.700 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [debug] | QUERY OK source="settings" db=0.2ms idle=317.3ms
SELECT s0."id", s0."onboarding", s0."pro_enabled", s0."yt_dlp_version", s0."apprise_version", s0."apprise_server", s0."youtube_api_key", s0."video_codec_preference", s0."audio_codec_preference" FROM "settings" AS s0 LIMIT 1 []
2024-11-27 18:49:52.702 request_id=73c07d05-2137-4199-9cf0-38ca8d5595be [info] | Sent 200 in 6ms
2024-11-27 18:50:28.216 request_id=515417d1-2043-4366-97bc-c1057f1b520b [info] | POST /sources
2024-11-27 18:50:28.217 request_id=515417d1-2043-4366-97bc-c1057f1b520b [debug] | Processing with PinchflatWeb.Sources.SourceController.create/2
  Parameters: %{"_csrf_token" => "MmQ7AgR1UiMvBnk0fAAZDHYhXjQFXwoZU7pvi91RVl5W8fWI3hnLw3kR", "source" => %{"custom_name" => "Fab Rats", "download_cutoff_date" => "", "download_media" => "true", "fast_index" => "true", "media_profile_id" => "1", "original_url" => "https://www.youtube.com/@FabRats", "output_path_template_override" => "", "retention_period_days" => "", "title_filter_regex" => ""}}
  Pipelines: [:browser]
2024-11-27 18:50:28.220 request_id=515417d1-2043-4366-97bc-c1057f1b520b [info] | [command_wrapper]: /usr/local/bin/yt-dlp called with: https://www.youtube.com/@FabRats --simulate --skip-download --ignore-no-formats-error --playlist-end 1 --print-to-file %(.{channel,channel_id,playlist_id,playlist_title,filename})j /tmp/pinchflat/data/85f0232da0de0eb03ec5d156cd2781f4fcc5e00ca2fe500ed764d5defa4db816.json --windows-filenames --quiet --cache-dir /tmp/pinchflat/data/yt-dlp-cache

```

**Comment by kieraneglin** [comment]
Hey there! Thanks for the report (:

I've seen this before, but it was due to a reverse proxy misconfiguration. What are the odds you're running the app behind a reverse proxy?

Also, I just released a new version earlier today which includes a bump to `yt-dlp` - would you mind upgrading to the latest and telling me if that fixes things?

**Comment by adamsih300u** [comment]
Thanks for your reply! I have re-set my manifest to load the latest image. `yt-dlp` shows as version 2024.11.18 when I exec into the into the container.

I DO have a reverse proxy running in my configuration, however I reconfigured my manifest to bypass that and use the LoadBalancer instead (which provides direct IP and I can connect to Pinchflat on port 8945). Now, when adding the source the Pinchflat page sits there awhile and I get a 'Connection Reset' notice in the browser - as opposed to the timeout when using the reverse proxy. The container itself does not appear to be restarting or anything, and no errors are thrown in (in my case) Kubernetes. I'll keep tinkering around with this too and see if I can shed any light on it.

Thanks!

**Comment by adamsih300u** [comment]
Tested the latest image in a completely vanilla Docker run using the command in the README, and it works fine. It takes several moments for the UI to update after submitting the new source, but it does refresh after a bit. I would feel confident this would work fine via Docker compose as well (which is how I used to run it before Kubernetes). I'll hunt around a bit to try to find some way of extending timeouts in my environment to see if I can't make it work, but I would say there's probably no issue here aside from environment mis-configurations.

**Comment by kieraneglin** [comment]
Thank you for following up! Feel free to re-open this issue or create a new one if you find anything else (:

**Comment by ajdunevent** [comment]
People who are smarter than I am: what are the repercussions of using host networking mode in the docker compose for pinchflat?  It was the only way I found to get around this issue.  If it isn't problematic, should it be suggested in the installation documentation as possibly needed?  If it is problematic, any pointers to help me troubleshoot my issue?  I was directly connecting to the IP and port of the host machine (running docker) without a proxy but getting a "the connection was reset" from Firefox every time I tried to add my first source.  Thanks to all!
